v83.0.0

jverkoey opened the pull request.

Commit:
[Dialogs] Fix Tall Font example (#7151)

Adds a background color (no longer completely transparent) to the example's
view and applies theming to the button and dialog.

|Before|After|
|----|----|
|![dialog-tall-before](https://user-images.githubusercontent.com/1753199/56212182-80a81180-6027-11e9-812c-ff92c52eb758.gif)|![dialog-tall-after](https://user-images.githubusercontent.com/1753199/56212189-856cc580-6027-11e9-85b1-cec11940cc90.gif)|

Follow-up to #7062
Closes #7148

andrewoverton committed:
[TextFields] Create concrete MDCFilledTextField and MDCOutlinedTextField classes based on the Contained Input View architecture (#7144)

In this PR I have created MDCFilledTextField and MDCOutlinedTextField concrete classes that inherit from an MDCInputTextField base class. There are no significant changes to how the text fields work under the hood--they are still contained input views. However, that protocol conformance is now private. If we were to include these classes in our library clients would not need to have any idea what a Contained Input View is. For discussion on the rationale for doing this see the ill-fated https://github.com/material-components/material-components-ios/pull/6985 and [b/130306935](http://b/130306935).
 
Closes #7111.

Commit:
Merge branch 'release-candidate' into develop

andrewoverton committed:
Update to Swift 4.2 (#7166)

This PR updates the Swift version to 4.2.

Partially resolves #6874.

nokeeo committed:
[BottomNavigation] Adds a large item dialog view (#7146)

Adds a view for displaying a large icon and title when the user has dynamic type enabled.

This is part of a larger effort to add dynamic type support to `MDCBottomNavigationBarController`.

Closes #7122

andrewoverton committed:
Add @objc annotations to get examples to show up in Dragons (#7168)

This is a follow up PR for #7166 adds @objc annotations to Swift catalogMetadata() methods, because the Swift 4 compiler no longer attempts to infer what methods should be visible to Objective-C. As a result of this change in the compiler, no Swift examples were showing up in Dragons after #7166. See this article: https://useyourloaf.com/blog/objc-warnings-upgrading-to-swift-4/ for additional context.

yarneo committed:
[ContainerScheme] Graduate ContainerScheme to ready. (#7170)

This PR graduates ContainerScheme to ready.

This includes updating the podspecs, podfile, all the import statements related to ContainerScheme, updating .kokoro rewrite rules, and finally the readme to not have ContainerScheme regarded to as being in beta.

Ran locally kokoro with -b bazel successfully.

Resolves #6732

andrewoverton committed:
Remove deprecated .swift-version file (#7171)

This PR gets rid of the .swift-version file. I guess the .swift-version file has been deprecated since Cocoapods 1.4. See the Cocoapods 1.4 release notes for additional context: https://blog.cocoapods.org/CocoaPods-1.4.0/

jverkoey committed:
[docs] Migrate the theming documentation to the docs folder. (#7173)

The documentation has been reviewed and copy-edited for consistency and accuracy.

Added a theming patterns section that describes a few patterns for theming in the context of an app.

Closes https://github.com/material-components/material-components-ios/issues/7074

fluxminds committed:
Add Container Scheme Unit Tests to scheme. (#7177)

Follow up to #6732.

Add missing Container's unit test target into scheme.

fluxminds committed:
Move Tabs theming extension to ready. (#7176)

closes https://github.com/material-components/material-components-ios/issues/7162

fluxminds committed:
[Cards] Move Cards theming extension to ready. (#7178)

closes https://github.com/material-components/material-components-ios/issues/7165

fluxminds committed:
[Chips] Move Chips Theming to Ready. (#7179)

closes https://github.com/material-components/material-components-ios/issues/7164

andrewoverton committed:
[TextFields] Graduate TextFields theming extension to ready (#7184)

This PR graduates the textfields theming extensions to ready.

Closes #7161.

yarneo committed:
Update theming.md (#7186)

Small improvement to the obj-c code snippet using Swift instead of Objc

andrewoverton committed:
[Buttons] Graduate Buttons+Theming to ready (#7187)

This PR graduates the buttons theming extension to ready.

Closes #7158.

codeman7 committed:
[AppBar]! Rename umbrella header for the theming extension (#7185)

## Context
In working on #7159 I noticed that the umbrella header for the _theming extension_ for [`MDCAppBar`](https://github.com/material-components/material-components-ios/tree/develop/components/AppBar) does not match our convention which is `Material<#componentName>+Theming.h`.

## The fix
Rename this file to use the convention outlined in the design doc. This was a replace of `MaterialTheming.h` -> `Theming.h`. This was done as a move operation and then the unit test were updated.

## Why breaking
This is a breaking change since we are renaming the umbrella header. This will break clients using swift and ObjectiveC. 

#### Swift

```diff
- import MaterialComponents.MaterialAppBar_MaterialTheming
+ import MaterialComponents.MaterialAppBar_Theming
```

#### ObjC

```diff
- #import "MaterialAppBar+MaterialTheming.h"
+ #import "MaterialAppBar+Theming.h"
```

codeman7 committed:
[ActionSheet] Remove dependency on helper methods in test. (#7183)

Context
Currently there is a helper method within the ActionSheet components test directory that is used across Podspec.

The fix
This PR removes that dependency across podspecs and updates the test accordingly. This work is related to #7181.

Bugs
Prework for #7160
Related to #6957

jverkoey committed:
[Buttons] Copy-edits on the theming documentation for clarity. (#7192)

Specifically:

- Removed a fair amount of duplicated text.
- Link to other articles where applicable.

Part of https://github.com/material-components/material-components-ios/issues/6592

fluxminds committed:
[Buttons] Add fallback behavior to setBorderColor API. (#7140)

closes https://github.com/material-components/material-components-ios/issues/3409

jverkoey committed:
[scripts] Fix the generate_readme script's jazzy symbol generation. (#7196)

Context is described in the associated issue.

The fix here is to force jazzy to only look at a given component when resolving headers. This means the generate_readme script will generate some import errors, so I've added a notice to the script indicating that these errors are likely ok.

Closes https://github.com/material-components/material-components-ios/issues/7195

jverkoey committed:
(automated) Regenerate all component readmes. (#7201)

This is an automated change. It was generated by running:

    ./scripts/generate_all_readmes

codeman7 committed:
[ActionSheet] Graduate theming extension to ready (#7181)

## Context
As part of our work to move away from _themer_ objects and onto _theming extension_ we need to graduate all of our components theming extensions to the MaterialComponents pod. 

## The fix
This PR updates the ActionSheet theming extension to the MaterialComponents pod, as well as updating examples and unit test. Additionally this removes the documentation for _themers_ since they were never out of the MaterialComponentsBeta podspec. The documentation for _themers_ is replaced with documentation for the _theming extension_.

## Bug
Closes #7160

codeman7 committed:
[Buttons] Fix broken link (#7198)

The link within the buttons theming extension documentation was broken, this resulted in a 404 error. The doc link was off by 1 directory, this changes updates the link to go up one level.

codeman7 committed:
[AppBar] Update documentation around theming (#7197)

As part of our move to theming extensions away from themers we need to update our documentation as well. This change removes all mention of "beta" within the theming-extension.md file as this extension is soon to go to the ready podspec. This also updates the older themer docs with the annotation "(To Be Deprecated)" as these are soon to be deprecated then removed. This work is related to #7159.

jverkoey committed:
[scripts] Fix relative link rewriting. (#7205)

Context is in the associated issue.

This affects the perl expression that prepends `docs/` to local urls. Specifically, it makes the perl expression only modify urls that link to a sibling markdown file composed of `\w` characters (e.g. `theming.md`). This is not perfect, but it addresses our needs for now. If/when we encounter other types of local urls that we can address them separately.

For now, this change fixes the incorrect behavior described in the associated bug.

Closes https://github.com/material-components/material-components-ios/issues/7203

jverkoey committed:
[ActivityIndicator] Annotate all themer APIs with the recommended new APIs. (#7211)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[Buttons] Copy-edit the documentation. (#7206)

This change annotates the button Themer documentation as "to be deleted" and edits the copy for conciseness.

Closes https://github.com/material-components/material-components-ios/issues/6592

jverkoey committed:
(docs/theming) Move "How to get the code" nearer to the top of the file. (#7210)

Installation and dependency steps are the first step required to use Theming extensions, so it's important that these steps are near the top of the documentation.

Follow-up for https://github.com/material-components/material-components-ios/issues/7074

jverkoey committed:
[AppBar] Annotate themer APIs with the recommended APIs. (#7213)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[BottomAppBar] Annotate themer APIs with the recommended APIs. (#7214)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[BottomNavigation] Annotate themer APIs with the recommended APIs. (#7215)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[BottomSheet] Annotate themer APIs with the recommended APIs. (#7216)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[ButtonBar] Annotate themer APIs with the recommended APIs. (#7217)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[Buttons] Annotate themer APIs with the recommended APIs. (#7218)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[Cards] Annotate themer APIs with the recommended APIs. (#7219)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[Chips] Annotate themer APIs with the recommended APIs. (#7220)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[Dialogs] Annotate themer APIs with the recommended APIs. (#7221)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[FeatureHighlight] Annotate themer APIs with the recommended APIs. (#7222)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[FlexibleHeader] Annotate themer APIs with the recommended APIs. (#7223)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[Ink] Annotate themer APIs with the recommended APIs. (#7225)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[HeaderStackView] Annotate themer APIs with the recommended APIs. (#7224)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[List] Annotate themer APIs with the recommended APIs. (#7226)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[NavigationDrawer] Annotate themer APIs with the recommended APIs. (#7229)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[NavigationBar] Annotate themer APIs with the recommended APIs. (#7228)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[PageControl] Annotate themer APIs with the recommended APIs. (#7230)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[ProgressView] Annotate themer APIs with the recommended APIs. (#7231)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[Slider] Annotate themer APIs with the recommended APIs. (#7232)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[Tabs] Annotate themer APIs with the recommended APIs. (#7234)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[Snackbar] Annotate themer APIs with the recommended APIs. (#7233)

Part of https://github.com/material-components/material-components-ios/issues/7108

jverkoey committed:
[TextFields] Annotate themer APIs with the recommended APIs. (#7235)

Part of https://github.com/material-components/material-components-ios/issues/7108

codeman7 committed:
[Docs] Fix links within documenation (#7208)

This change fixes the broken links in MDCActionSheet and MDCAppBar components. This also moves the Themer documentation to an "unsupported" section.

codeman7 committed:
[Docs] Clean up theming doc (#7191)

This PR will help clients who are using our library what to import if they are using our theming extensions. This also cleans up the ObjectiveC code to use more modern ObjC style.

andrewoverton committed:
[Dialogs, TextFields] Update docs (#7209)

#7187 and #7184 graduated Buttons and TextFields theming extensions to ready, but didn't update the docs. Similarly, #7190, which has not yet been merged in, is not updating the docs to reflect the graduation of the Dialogs theming extension to ready.

This PR updates docs for Dialogs and TextFields theming extensions to reflect them being marked ready. Once #7190 is merged in the Dialogs changes here will be accurate.

I will address the out-of-date Buttons docs once #7206 is merged in.

codeman7 committed:
[AppBar] Graduate theming extension to ready (#7182)

## Context
As part of our work to move away from themer objects and onto theming extension we need to graduate all of our components theming extensions to the MaterialComponents pod.

## The fix
This PR updates the AppBar theming extension to the MaterialComponents pod, as well as updating examples and unit test.

jverkoey committed:
[Buttons] Fix theming headers and copy. (#7236)

Fixed the headers throughout the content.

Adjusted the "typical use" copy to call out that MDCButtons should be initialized with alloc/init.

Follow-up to https://github.com/material-components/material-components-ios/issues/6592

Commit:
Automatic changelog preparation for release.

Commit:
Update release notes.

Commit:
Bump the release number to 83.0.0.

jverkoey commented:
@bernaferrari

> I have no idea what is this:
https://github.com/material-components/material-components-ios/blob/develop/docs/theming.md#how-to-get-the-code-1

Fair point, those docs look pretty crufty. They're part of our unsupported Themer APIs documentation though so it's unlikely that we'll devote time toward cleaning it up.

bernaferrari commented:
No problem. Examples is not in the same section as the deprecated themer API. That's what created confusion for me. I thought they were examples for the supported theme, because, well, different section.

jverkoey commented:
Ah; yep more headers to be fixed :) If you're interested in sending us PRs that would be an easy one for you to start with!

bernaferrari commented:
Thanks, #7240.

codeman7 commented on CHANGELOG.md:
```diff
@@ -1,3 +1,931 @@
+# 83.0.0
+
+This major release graduates several Theming extensions from Beta and includes a breaking change for
```
_Tip:_ "Theming Extensions" or "theming extensions"

codeman7 commented on CHANGELOG.md:
```diff
@@ -1,3 +1,931 @@
+
+This major release graduates several Theming extensions from Beta and includes a breaking change for
+AppBar's imports. All Themer APIs have been annotated as "to be deprecated" in lieu of Theming
+extensions.
```
How do you feel about listing all the components that are graduating their theming extension?

jverkoey replied:
Done.

codeman7 commented:
Should we pull in #7241?

yarneo committed:
[Catalog] Add @objc annotations to our containerScheme instances in Swift (#7243)

We need to add @objc annotations to the containerScheme instances in our Swift examples, because we moved to Swift 4.2, the respondsToSelector won't find the setContainerScheme: setter otherwise.

andrewoverton committed:
[Catalog, Dragons] Fix schemes (#7241)

My MDCDragons scheme was messed up on develop after running pod install. I checked with Cody and his was too. I was seeing this:

<img width="892" alt="Screen Shot 2019-04-22 at 5 01 56 PM" src="https://user-images.githubusercontent.com/8020010/56530476-6dfa7480-6520-11e9-8c36-832daa3bb4db.png">

The problem seems to have been resolved by reverting the last changes made to the scheme files and simply opening the scheme editor and letting Xcode automatically make the changes it wanted. I made sure that all of the test targets were included in both schemes afterward.

Closes #7242.

Commit:
Add list of graduated theming extensions.

Commit:
Casing.

romoore commented:
@jverkoey Is this actually ready for review? I see active work being done on the PR.

jverkoey commented:
@romoore Yes it is actively being worked on in response to review feedback.

yarneo committed:
update (#7246)

We need to add @objc annotations to colorScheme and typographyScheme instances in our Swift examples, because we moved to Swift 4.2, the respondsToSelector won't find `setColorScheme:` and `setTypographyScheme:` setters otherwise.
